pkgs/gnomod/gnomod.go

```diff
@@ -0,0 +1,210 @@
+package gnomod
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/pkgs/gnolang"
+	"github.com/gnolang/gno/pkgs/std"
+	"golang.org/x/mod/modfile"
+	"golang.org/x/mod/module"
+)
+
+var (
+	source        = "test3.gno.land:36657" // "127.0.0.1:26657"
+	queryPathFile = "vm/qfile"
+)
+
+// ReadModFile reads, parses and validates the mod file at gnomod.
+func ReadModFile(absModPath string) (f *File, err error) {
+	data, err := os.ReadFile(absModPath)
+	if err != nil {
+		return nil, fmt.Errorf("reading gno.mod: %s", err)
+	}
+
+	f, err = Parse(absModPath, data)
+	if err != nil {
+		return nil, fmt.Errorf("parsing gno.mod: %s", err)
+	}
+
+	return f, err
+}
+
+func IsModFileExist(absModPath string) bool {
+	_, err := os.Stat(absModPath)
+	return err == nil
+}
+
+// GetGnoModPath returns the path for gno modules
+func GetGnoModPath() (string, error) {
+	goPath := os.Getenv("GOPATH")
+	if goPath == "" {
+		return "", errors.New("GOPATH not found")
+	}
+
+	return filepath.Join(goPath, "pkg/gnomod"), nil
+}
+
+// FetchModPackages fetches and writes gno.mod packages
+// in GOPATH/pkg/gnomod/
+func FetchModPackages(f *File) error {
+	gnoModPath, err := GetGnoModPath()
+	if err != nil {
+		return fmt.Errorf("fetching mods: %s", err)
+	}
+
+	if f.Require != nil {
+		for _, r := range f.Require {
```

Comment:
```suggestion
		for _, r := range f.Require {
```

this works and is more readable

Reply:
Resolved-by: 4ecb4ed